src/annotator/frame-observer.js

```diff
@@ -120,15 +120,11 @@ export function onDocumentReady(iframe) {
     // on Chrome and Safari, and an 'uninitialized' readyState on Firefox and
     // (3) has a empty body and head. If a blank document is detected and there
     // is a 'src' attribute, it is expected that the blank document will be
-    // replaced by the document from 'src'.
+    // replaced by the final document.
     /* istanbul ignore next */
     if (
-      // @ts-ignore
-      state === 'uninitialized' ||
-      (state === 'complete' &&
-        iframe.hasAttribute('src') &&
-        iframeDocument.head.hasChildNodes() === false &&
-        iframeDocument.body.hasChildNodes() === false)
+      iframeDocument.location.href === 'about:blank' &&
+      iframe.hasAttribute('src')
```

Comment:
`src` could be `about:blank` in which this logic will fail. I don't see any logic for somebody using `src` in that way.

Reply:
> I don't see any logic for somebody using src in that way.

We could check for a `src` value that is different from `about:blank`. I wonder what happens if the `src` value is a `data:` or `blob:` URL that can be resolved synchronously?